lib/service/connect.go

```diff
@@ -600,41 +600,57 @@ type rotationStatus struct {
 	ca services.CertAuthority
 }
 
-// rotate is called to check if rotation should be triggered.
-func (process *TeleportProcess) rotate(conn *Connector, localState auth.StateV2, remote services.Rotation) (*rotationStatus, error) {
-	id := conn.ClientIdentity.ID
-	local := localState.Spec.Rotation
+// checkPrincipals returns a boolean that indicates the host certificate
+// needs to be regenerated.
+func checkPrincipals(conn *Connector, additionalPrincipals []string, dnsNames []string) bool {
+	var principalsChanged bool
+	var dnsNamesChanged bool
 
-	additionalPrincipals, dnsNames, err := process.getAdditionalPrincipals(id.Role)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	// If the list of principals contains 0.0.0.0 (meaning advertise_ip has not
+	// been set), remove it from the list of principals. The 0.0.0.0 values tells the
+	// auth server to "guess" the nodes IP. Since this IP is not known (possibly
+	// even to the node itself if it's behind some kind of NAT), checking if it's
+	// in the list of principals would not make sense.
+	principalToCheck := additionalPrincipals
```

Comment:
can we just always non-conditionally exclude 0.0.0.0 from the slice, would not it make the logic simpler?